Add c.network.list()

mbojan opened the pull request:
c.f. #603

mbojan committed:
Add c.network.list()

mbojan committed:
Add c.network.list.R to Collate field

mbojan committed:
Fix attribute checking

krivit commented on R/c.network.list.R:
```diff
@@ -0,0 +1,61 @@
+c.network.list <- function(..., check_attr = TRUE) {
+  dots <- list(...)
+  
+  # Check a list elements using all.equal or identical
+  check_list <- function(lst, fun = all.equal) {
+    r <- vapply(lst[-1], function(x) isTRUE(fun(x, lst[[1]])), logical(1))
+    all(r)
+  }
```
This function is similar to `statnet.common::all_identical()`. Would you mind contributing it (or a version of it) there? Also, it might be worth benchmarking, because the `all_identical()`'s `for` loop seems to perform about 4 times faster.

mbojan committed:
Use statnet.common::all_identical()

Uses the new variant of all_identical() allowing arbitary
pairwise-comparison function.

krivit commented:
Actually, could you please put the `c.network.list()` into `R/network.list.R`?

mbojan committed:
Move c.network.list() to network.list.R

mbojan committed:
Drop old file from Collate

krivit commented:
@mbojan , I've collapsed the changes to avoid having to merge with a file that no longer exists. One last item is perhaps a unit test or two?

krivit commented:
> Ok, will do. I was wondering where the conflict came from.

I went through and merged a bunch of files in the `R` directory, which included all the files pertaining to `network.list` and its methods. In order to keep the version history, I needed to do it in a roundabout way (essentially create a branch for each source file, rename it to a temporary file in that branch, then merge the branches onto each other, and rename the temporary file (which will now contain all the files' contents) into its final name).

mbojan committed:
Use .p not fun argument

mbojan committed:
Test c.network.list()

mbojan committed:
Export c.network.list() method

mbojan committed:
Add c.network.list() example

mbojan committed:
Roxygenize

krivit committed:
Simplified some of the code in c.network.list() and fixed formatting.

krivit commented on R/network.list.R:
```diff
@@ -91,3 +91,72 @@ summary.network.list <- function (object, stats.print=TRUE,
+    control = attr(dots[[1]], "control"),
+    response = attr(dots[[1]], "response"),
+    stats = structure(
+      do.call("rbind", map(dots, ~attr(.x, "stats"))),
```
This does not work for `simulate(simplify = FALSE)`, because then `attr(., "stats")` is an `mcmc.list`, not a matrix.

krivit commented on tests/testthat/test-network.list.R:
```diff
@@ -0,0 +1,42 @@
+suppressMessages({
+  data("faux.desert.high", package = "ergm")
+  fit1 <- ergm(
+    faux.desert.high ~ mm("sex"),
+    control = control.ergm(
+      force.main=TRUE,
+      MCMLE.maxit = 1,
+      seed = 1
+    )
+  )
+  fit2 <- ergm(
+    faux.desert.high ~ mm("sex"),
+    control = control.ergm(
+      force.main=TRUE,
+      MCMLE.maxit = 1,
+      seed = 2
+    )
+  )
```
Can we just hard-code the coefficients in the subsequent `simulate()` calls?

mbojan committed:
Simplify tests of c.network.list
